src/cargo/core/dependency.rs

```diff
@@ -327,14 +339,14 @@ impl Dependency {
         self.inner.matches_id(id)
     }
 
-    pub fn map_source(self, to_replace: &SourceId, replace_with: &SourceId)
+    pub fn map_source(mut self, to_replace: &SourceId, replace_with: &SourceId)
```

Comment:
Semantically we want `map_source(&self) -> Cow<Dependency>` here, but I don't think that would be an actual improvement performance or readability wise.